### ISSUE TITLE
mobilenet_v2 not working with the newest version(1.1.0)

### ISSUE THREAD

**Issue by egeozsoy** [pytorch/pytorch#23487]
## 🐛 Bug

I am getting a segmentation error when I call loss.backward() while using mobilenet_v2.

## To Reproduce

```
import torch
import torchvision

model  = torchvision.models.mobilenet_v2(num_classes=2)

random_example = torch.rand(size=(4,3,224,224))
random_labels = torch.LongTensor([0,1,0,1])

output = model(random_example)

loss_function = torch.nn.NLLLoss()
loss = loss_function(output,random_labels)
loss.backward()
print('Completed')
```
I am receiving a Segmentation fault: 11 when I run this code. If I use a very small tensor like 
torch.rand(size=(4,3,32,32)) I am not getting this error. (Also I believe that torch==1.0.0 doesn't have this bug)

## Expected behavior
It should just return 'Completed'.

## Environment

PyTorch version: 1.1.0.post2
Is debug build: No
CUDA used to build PyTorch: None

OS: Mac OSX 10.14.4
GCC version: Could not collect
CMake version: version 3.10.2

Python version: 3.6
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip3] numpy==1.17.0
[pip3] torch==1.1.0.post2
[pip3] torchvision==0.3.0
[conda] Could not collect

## Additional context
I have 32 GB of ram.

**Comment by egeozsoy** [comment]
I think I was able to locate the source of the bug. 
It is coming from the InvertedResidual class, and more specifically, in my case it started working when I commented out  "nn.BatchNorm2d(oup)". Obviously this isn't a real solution but maybe it can help.
```
class InvertedResidual(nn.Module):
    def __init__(self, inp, oup, stride, expand_ratio):
        super(InvertedResidual, self).__init__()
        self.stride = stride
        assert stride in [1, 2]

        hidden_dim = int(round(inp * expand_ratio))
        self.use_res_connect = self.stride == 1 and inp == oup

        layers = []
        if expand_ratio != 1:
            # pw
            layers.append(ConvBNReLU(inp, hidden_dim, kernel_size=1))
        layers.extend([
            # dw
            ConvBNReLU(hidden_dim, hidden_dim, stride=stride, groups=hidden_dim),
            # pw-linear
            nn.Conv2d(hidden_dim, oup, 1, 1, 0, bias=False),
            nn.BatchNorm2d(oup),
        ])
        self.conv = nn.Sequential(*layers)

    def forward(self, x):
        if self.use_res_connect:
            return x + self.conv(x)
        else:
            return self.conv(x)
```

**Comment by egeozsoy** [comment]
I was able to confirm that it works using Google Colab(Linux). So it might be mac specific.

**Comment by soumith** [comment]
similar to / same as https://github.com/pytorch/pytorch/issues/23487

**Comment by colesbury** [comment]
@fmassa can you look into this?

**Comment by fmassa** [comment]
I've just tried running the above script in my Mac, and everything worked as expected.

```python
In [34]: import torch
    ...: import torchvision
    ...:
    ...: model  = torchvision.models.mobilenet_v2(num_classes=2)
    ...:
    ...: random_example = torch.rand(size=(4,3,224,224))
    ...: random_labels = torch.LongTensor([0,1,0,1])
    ...:
    ...: output = model(random_example)
    ...:
    ...: loss_function = torch.nn.NLLLoss()
    ...: loss = loss_function(output,random_labels)
    ...: loss.backward()
    ...: print('Completed')
    ...:
Completed

In [35]: print(torch.__version__, torchvision.__version__)
1.1.0.dev20190520 0.3.0a0+68ac54e
```

**Comment by fmassa** [comment]
I'm suspecting that there is an OOM happening somewhere maybe?

**Comment by egeozsoy** [comment]
I have torch version: 1.1.0.post2 and torchvision version: 0.3.0. Do you think that can be related?

**Comment by fmassa** [comment]
Could you try updating to the latest nightly for PyTorch / torchvision?

**Comment by egeozsoy** [comment]
Yes I will install and try again, meanwhile if I use a smaller tensor size such as torch.rand(size=(4,3,32,32)) it works. This indicates to an OOM problem. But I have more than 16 GB free memory available, and am also able to run other models such as Resnet50 without problems

**Comment by fmassa** [comment]
can you try running the code on gdb?

```
gdb python
> run my_script.py
# it will segfault, and then do
> bt
```
and paste the log here?

**Comment by egeozsoy** [comment]
I have succesfully installed nightly release of torch, but what is the easiest way to install nightly release of torchvision? There is no pip install torchvision_nightly right?

**Comment by fmassa** [comment]
No, there isn't a torchvision nightly as of today, so you'll need to clone the repo and install it from source.

Don't forget to uninstall torchvision first, and then do
```python
pip uninstall torchvision
python setup.py install
```

**Comment by egeozsoy** [comment]
Okey installing from source will take some time, so I am first trying to run it with gdb. I just installed gdb with `brew install gdb`, and having some difficulties using it.

```gdb python
> run my_script.py
Starting program:  my_script.by
No executable file specified.
Use the "file" or "exec-file" command.
```

Does it sound familiar?

**Comment by fmassa** [comment]
there is a typo somewhere?
```
> run my_script.py
Starting program:  my_script.by
                            ^^^ ??
```

**Comment by colesbury** [comment]
@egeozsoy: use something like `run python my_script.py` or alternatively `gdb --args python my_script.py` from the command line.

**Comment by egeozsoy** [comment]
Typo was just while manually changing the script name for posting here, not while actually running it

**Comment by egeozsoy** [comment]
running with` run python my_script.py` produces the same error as before, and `gdb --args python my_script.py` produces a different error:

`not in executable format: file format not recognized`

**Comment by egeozsoy** [comment]
Okey so I am really not an expert in this but after some research I found that I can use lldb, which has a similar interface to gdb. And it works.

my_script.py crashed and afterwards I did `bt`

```
Process 78137 stopped
* thread #7, stop reason = EXC_BAD_ACCESS (code=1, address=0x12ddb0000)
    frame #0: 0x000000012d95d0b4
->  0x12d95d0b4: vmovups 0x40(%r9), %ymm7
    0x12d95d0ba: vfmadd231ps %ymm1, %ymm5, %ymm2       ; ymm2 = (ymm5 * ymm1) + ymm2
    0x12d95d0bf: vfmadd231ps %ymm1, %ymm6, %ymm3       ; ymm3 = (ymm6 * ymm1) + ymm3
    0x12d95d0c4: vmovups %ymm2, (%r13)
Target 0: (python) stopped.
(lldb) bt
* thread #7, stop reason = EXC_BAD_ACCESS (code=1, address=0x12ddb0000)
  * frame #0: 0x000000012d95d0b4
    frame #1: 0x0000000119f88ffe libcaffe2.dylib`mkldnn::impl::cpu::_jit_uni_dw_convolution_bwd_weights_t<(mkldnn::impl::cpu::cpu_isa_t)3>::execute_backward_weights() const + 830
    frame #2: 0x0000000119f88c9c libcaffe2.dylib`mkldnn::impl::cpu::_jit_uni_dw_convolution_bwd_weights_t<(mkldnn::impl::cpu::cpu_isa_t)3>::execute(mkldnn::impl::event_t*) const + 28
    frame #3: 0x0000000119b4b677 libcaffe2.dylib`mkldnn::impl::cpu::cpu_engine_t::submit(mkldnn_primitive*, mkldnn::impl::event_t*, mkldnn::impl::nstl::vector<mkldnn::impl::event_t*>&) + 119
    frame #4: 0x0000000119b3f302 libcaffe2.dylib`mkldnn::impl::stream_eager_t::submit_impl(unsigned long, unsigned long, mkldnn_primitive**) + 754
    frame #5: 0x0000000119b3e962 libcaffe2.dylib`mkldnn_stream::submit(mkldnn::impl::nstl::vector<mkldnn_primitive*> const&, mkldnn_primitive**) + 274
    frame #6: 0x0000000119b3ed37 libcaffe2.dylib`mkldnn_stream_submit + 167
    frame #7: 0x00000001182ccf94 libcaffe2.dylib`mkldnn::stream::submit(std::__1::vector<mkldnn::primitive, std::__1::allocator<mkldnn::primitive> >) + 228
    frame #8: 0x00000001182d0c19 libcaffe2.dylib`at::native::mkldnn_convolution_backward_weights(c10::ArrayRef<long long>, at::Tensor const&, at::Tensor const&, c10::ArrayRef<long long>, c10::ArrayRef<long long>, c10::ArrayRef<long long>, long long, bool) + 6425
    frame #9: 0x000000011850f541 libcaffe2.dylib`at::TypeDefault::mkldnn_convolution_backward_weights(c10::ArrayRef<long long>, at::Tensor const&, at::Tensor const&, c10::ArrayRef<long long>, c10::ArrayRef<long long>, c10::ArrayRef<long long>, long long, bool) const + 353
    frame #10: 0x00000001182d21cf libcaffe2.dylib`at::native::mkldnn_convolution_backward(at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::ArrayRef<long long>, c10::ArrayRef<long long>, c10::ArrayRef<long long>, long long, std::__1::array<bool, 3ul>) + 591
    frame #11: 0x000000011850f755 libcaffe2.dylib`at::TypeDefault::mkldnn_convolution_backward(at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::ArrayRef<long long>, c10::ArrayRef<long long>, c10::ArrayRef<long long>, long long, std::__1::array<bool, 3ul>) const + 341
    frame #12: 0x00000001226eb8b4 libtorch.1.dylib`torch::autograd::VariableType::mkldnn_convolution_backward(at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::ArrayRef<long long>, c10::ArrayRef<long long>, c10::ArrayRef<long long>, long long, std::__1::array<bool, 3ul>) const + 3620
    frame #13: 0x00000001223781c5 libtorch.1.dylib`torch::autograd::generated::MkldnnConvolutionBackward::apply(std::__1::vector<torch::autograd::Variable, std::__1::allocator<torch::autograd::Variable> >&&) + 517
    frame #14: 0x00000001222bb2ef libtorch.1.dylib`torch::autograd::Function::operator()(std::__1::vector<torch::autograd::Variable, std::__1::allocator<torch::autograd::Variable> >&&) + 511
    frame #15: 0x00000001222b2da1 libtorch.1.dylib`torch::autograd::Engine::evaluate_function(torch::autograd::FunctionTask&) + 1377
    frame #16: 0x00000001222b2433 libtorch.1.dylib`torch::autograd::Engine::thread_main(torch::autograd::GraphTask*) + 275
    frame #17: 0x00000001222b2216 libtorch.1.dylib`torch::autograd::Engine::thread_init(int) + 278
    frame #18: 0x00000001179e2b2c libtorch_python.dylib`torch::autograd::python::PythonEngine::thread_init(int) + 44
    frame #19: 0x00000001222c14e2 libtorch.1.dylib`void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (torch::autograd::Engine::*)(int), torch::autograd::Engine*, int> >(void*) + 66
    frame #20: 0x0000000101c2b2eb libsystem_pthread.dylib`_pthread_body + 126
    frame #21: 0x0000000101c2e249 libsystem_pthread.dylib`_pthread_start + 66
    frame #22: 0x0000000101c2a40d libsystem_pthread.dylib`thread_start + 13
```

**Comment by egeozsoy** [comment]
* thread #7, stop reason = EXC_BAD_ACCESS (code=1, address=0x12ddb0000)
  * frame #0: 0x000000012d95d0b4
    frame #1: 0x0000000119f88ffe libcaffe2.dylib`mkldnn::impl::cpu::_jit_uni_dw_convolution_bwd_weights_t<(mkldnn::impl::cpu::cpu_isa_t)3>::execute_backward_weights() const + 830
    frame #2: 0x0000000119f88c9c libcaffe2.dylib`mkldnn::impl::cpu::_jit_uni_dw_convolution_bwd_weights_t<(mkldnn::impl::cpu::cpu_isa_t)3>::execute(mkldnn::impl::event_t*) const + 28
    frame #3: 0x0000000119b4b677 libcaffe2.dylib`mkldnn::impl::cpu::cpu_engine_t::submit(mkldnn_primitive*, mkldnn::impl::event_t*, mkldnn::impl::nstl::vector<mkldnn::impl::event_t*>&) + 119
    frame #4: 0x0000000119b3f302 libcaffe2.dylib`mkldnn::impl::stream_eager_t::submit_impl(unsigned long, unsigned long, mkldnn_primitive**) + 754
    frame #5: 0x0000000119b3e962 libcaffe2.dylib`mkldnn_stream::submit(mkldnn::impl::nstl::vector<mkldnn_primitive*> const&, mkldnn_primitive**) + 274
    frame #6: 0x0000000119b3ed37 libcaffe2.dylib`mkldnn_stream_submit + 167
    frame #7: 0x00000001182ccf94 libcaffe2.dylib`mkldnn::stream::submit(std::__1::vector<mkldnn::primitive, std::__1::allocator<mkldnn::primitive> >) + 228
    frame #8: 0x00000001182d0c19 libcaffe2.dylib`at::native::mkldnn_convolution_backward_weights(c10::ArrayRef<long long>, at::Tensor const&, at::Tensor const&, c10::ArrayRef<long long>, c10::ArrayRef<long long>, c10::ArrayRef<long long>, long long, bool) + 6425
    frame #9: 0x000000011850f541 libcaffe2.dylib`at::TypeDefault::mkldnn_convolution_backward_weights(c10::ArrayRef<long long>, at::Tensor const&, at::Tensor const&, c10::ArrayRef<long long>, c10::ArrayRef<long long>, c10::ArrayRef<long long>, long long, bool) const + 353
    frame #10: 0x00000001182d21cf libcaffe2.dylib`at::native::mkldnn_convolution_backward(at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::ArrayRef<long long>, c10::ArrayRef<long long>, c10::ArrayRef<long long>, long long, std::__1::array<bool, 3ul>) + 591
    frame #11: 0x000000011850f755 libcaffe2.dylib`at::TypeDefault::mkldnn_convolution_backward(at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::ArrayRef<long long>, c10::ArrayRef<long long>, c10::ArrayRef<long long>, long long, std::__1::array<bool, 3ul>) const + 341
    frame #12: 0x00000001226eb8b4 libtorch.1.dylib`torch::autograd::VariableType::mkldnn_convolution_backward(at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::ArrayRef<long long>, c10::ArrayRef<long long>, c10::ArrayRef<long long>, long long, std::__1::array<bool, 3ul>) const + 3620
    frame #13: 0x00000001223781c5 libtorch.1.dylib`torch::autograd::generated::MkldnnConvolutionBackward::apply(std::__1::vector<torch::autograd::Variable, std::__1::allocator<torch::autograd::Variable> >&&) + 517
    frame #14: 0x00000001222bb2ef libtorch.1.dylib`torch::autograd::Function::operator()(std::__1::vector<torch::autograd::Variable, std::__1::allocator<torch::autograd::Variable> >&&) + 511
    frame #15: 0x00000001222b2da1 libtorch.1.dylib`torch::autograd::Engine::evaluate_function(torch::autograd::FunctionTask&) + 1377
    frame #16: 0x00000001222b2433 libtorch.1.dylib`torch::autograd::Engine::thread_main(torch::autograd::GraphTask*) + 275
    frame #17: 0x00000001222b2216 libtorch.1.dylib`torch::autograd::Engine::thread_init(int) + 278
    frame #18: 0x00000001179e2b2c libtorch_python.dylib`torch::autograd::python::PythonEngine::thread_init(int) + 44
    frame #19: 0x00000001222c14e2 libtorch.1.dylib`void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (torch::autograd::Engine::*)(int), torch::autograd::Engine*, int> >(void*) + 66
    frame #20: 0x0000000101c2b2eb libsystem_pthread.dylib`_pthread_body + 126
    frame #21: 0x0000000101c2e249 libsystem_pthread.dylib`_pthread_start + 66
    frame #22: 0x0000000101c2a40d libsystem_pthread.dylib`thread_start + 13

```

**Comment by egeozsoy** [comment]
Sorry pressed accidentally on close.

**Comment by fmassa** [comment]
This is an issue with MKLDNN.

**Comment by egeozsoy** [comment]
Do you still think I should install the latest nightly versions? If you think it might be useful, I will gladly do it

**Comment by soumith** [comment]
@egeozsoy it would be great if you test it on the nightly version and report back. That would tell us the path to take.

tagging @gujinghui @vpirogov @Jianhui-Li @jgong5 @uyongw . Considering a PyTorch release being prepared this week, please take a look at the segfault stemming from MKL-DNN

**Comment by egeozsoy** [comment]
I can't reproduce this using torch: 1.2.0.dev20190729  and torchvision: 0.3.0a0+2287c8f(both nightly releases)

**Comment by soumith** [comment]
great, thanks. this will be parts of the upcoming release next week.

@gujinghui and team, nothing to fix, looks like already fixed in a newer MKL-DNN release.

**Comment by egeozsoy** [comment]
Thanks for the quick responses!
